src_c/mixer.c

```diff
@@ -1918,16 +1926,14 @@ pgSound_New(Mix_Chunk *chunk)
 static PyObject *
 pgChannel_New(int channelnum)
 {
-    pgChannelObject *chanobj;
-
-    if (channelnum < 0 || channelnum >= Mix_GroupCount(-1))
-        return RAISE(PyExc_IndexError, "invalid channel index");
-
-    chanobj = PyObject_New(pgChannelObject, &pgChannel_Type);
-    if (!chanobj)
+    pgChannelObject *chanobj = PyObject_New(pgChannelObject, &pgChannel_Type);
+    if (!chanobj) {
         return NULL;
-
-    chanobj->chan = channelnum;
+    }
+    if (channel_init(chanobj, Py_BuildValue("(i)", channelnum), NULL)) {
+        PyObject_Free(chanobj);
```

Comment:
2 things here, the return of `Py_BuildValue` functioncall is leaked here, because `channel_init` only borrows a reference. Secondly, IG `Py_DECREF` should be used instead of `PyObject_Free` here, because that is the most commonly used API while cleaning a python object

I suggest having a `_channel_init` that takes a C int, so that both `channel_init` and `pgChannel_New` can call it